File: java-spring-boot-maven-hello/src/main/java/com/example/loanhello/config/ServiceConfig.java
package com.example.loanhello.config;

import org.springframework.beans.factory.annotation.Value;
import org.springframework.context.annotation.Configuration;

@Configuration
public class ServiceConfig {

    @Value("${service.name:loan-service}")
    private String name;

    @Value("${service.version:1.2.0}")
    private String version;

    @Value("${service.environment:production}")
    private String environment;

    public String getName() { return name; }
    public String getVersion() { return version; }
    public String getEnvironment() { return environment; }
}

File: kotlin-spring-boot-hello/src/main/kotlin/com/example/loanhello/config/ServiceConfig.kt
package com.example.loanhello.config

import org.springframework.boot.context.properties.ConfigurationProperties

@ConfigurationProperties(prefix = "service")
data class ServiceConfig(
    val name: String = "loan-service",
    val version: String = "1.2.0",
    val environment: String = "production"
)
